runtime/near-vm-runner/src/preload.rs

```diff
@@ -12,13 +12,27 @@ use near_vm_logic::types::PromiseResult;
 use near_vm_logic::{External, ProtocolVersion, VMConfig, VMContext, VMKind, VMOutcome};
 
 use crate::cache;
+use crate::memory::WasmerMemory;
 use crate::preload::VMModule::{Wasmer0, Wasmer1};
-use crate::wasmer1_runner::{default_wasmer1_store, run_wasmer1_module};
+use crate::wasmer1_runner::{default_wasmer1_store, run_wasmer1_module, Wasmer1Memory};
 use crate::wasmer_runner::run_wasmer0_module;
 
+const SHARE_MEMORY_INSTANCE: bool = false;
+
 enum VMModule {
     Wasmer0(wasmer_runtime::Module),
-    Wasmer1(wasmer::Module, wasmer::Store),
+    Wasmer1(wasmer::Module),
+}
+
+enum VMDataPrivate {
+    Wasmer0(Option<WasmerMemory>),
+    Wasmer1(Option<Wasmer1Memory>),
+}
+
+#[derive(Clone)]
+enum VMDataShared {
```

Comment:
An alternative is to "transpose" these two:

```rust
enum VMData {
  Wasmer0 { private: Option<WasmerMemory>, shared: () }
  Wasmer1 { private: Option<Wasmer1Memory>, shared: wasmer::Store }
}
```

Reply:
Well, for me it was rather important to distinguish shared vs. non-shared state, as they have different concurrent access policies.